src/Uno.UI/UI/Xaml/Controls/Image/Image.cs

```diff
@@ -251,218 +248,211 @@ public override string ToString()
-					var desiredSize = new Size();
-					switch (img.Stretch)
-					{
-						case Stretch.Uniform:
-							//If sourceSize is empty, aspect ratio is undefined so we return 0
-							// Since apsect ratio can have a lot of decimal, iOS ceils Image size to 0.5 if it's not a precise size (like 111.111111111)
-							// so the desiredSize will never match the actual size causing an infinite measuring and can freeze the app
-							desiredSize.Width = sourceSize == default(Size) ? 0 : Math.Ceiling(knownHeight * aspectRatio * 2) / 2;
-							desiredSize.Height = knownHeight;
-							break;
-						case Stretch.None:
-							desiredSize.Width = sourceSize.Width;
-							desiredSize.Height = sourceSize.Height;
-							break;
-						case Stretch.Fill:
-						case Stretch.UniformToFill:
-							desiredSize.Width = double.IsInfinity(constrainedAvailableSize.Width) ? sourceSize.Width : constrainedAvailableSize.Width;
-							desiredSize.Height = knownHeight;
-							break;
-					}
-
-					if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
-					{
-						this.Log().Debug(Panel.ToString() + $" measuring with knownHeight={knownHeight} with availableSize={constrainedAvailableSize}, returning desiredSize={desiredSize}");
-					}
 
-					return desiredSize;
+				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				{
+					this.Log().Debug(ToString() + $" measuring with knownHeight={knownHeight} with availableSize={constrainedAvailableSize}, returning desiredSize={desiredSize}");
 				}
 
-				throw new InvalidOperationException("Should never reach here.");
+				return desiredSize;
 			}
 
-			protected override Size ArrangeOverride(Size finalSize)
+			throw new InvalidOperationException("Should never reach here.");
+		}
+
+		protected override Size ArrangeOverride(Size finalSize)
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
 			{
-				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
-				{
-					this.Log().Debug(Panel.ToString() + $" arranging with finalSize={finalSize}");
-				}
+				this.Log().Debug(ToString() + $" arranging with finalSize={finalSize}");
+			}
 
-				//If we are given a non-zero size to draw into, set the target dimensions to load the image with accordingly
-				ImageControl.UpdateArrangeSize(finalSize);
-				ImageControl.SetTargetImageSize(finalSize);
+			//If we are given a non-zero size to draw into, set the target dimensions to load the image with accordingly
+			UpdateArrangeSize(finalSize);
+			SetTargetImageSize(finalSize);
 
-				if (this.Log().IsEnabled(LogLevel.Warning))
+			if (this.Log().IsEnabled(LogLevel.Warning))
+			{
+				if (_openedImage != null)
 				{
-					if (ImageControl._openedImage != null)
-					{
-						var renderedSize = finalSize.LogicalToPhysicalPixels();
-						var loadedSize = ImageControl.SourceImageSize.LogicalToPhysicalPixels();
+					var renderedSize = finalSize.LogicalToPhysicalPixels();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'finalSize' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5554394)